enhancements/ingress/default-ingress-controller.md

```diff
@@ -0,0 +1,110 @@
+This proposal eliminates the operator/administrator contention by building on the [User Defined IngressControllers](user-defined-default-ingress-controller.md) capability. Specifically, by:
+
+1. Changing the ingress-operator to stop creating ingresscontrollers altogether
+2. Changing the installer to always render a `default` IngressController
+
+These changes make explicit the notion that the ingress-operator is responsible only for managing IngressControllers created by end users, _not_ for creating IngressControllers.
+
+End users can now assume that the _only_ way IngressControllers are created are via the installer or through their own API calls.
+
+## Motivation
+
+The operator today has a confusing relationship to the `default` IngressController resource. Not only does the operator confound automation in its competition with users for ownership, but the operator is not even guaranteed to have the context necessary to restore any IngressController that was deleted, including the `default` instance. The operator should have a clear responsibility to manage resources brought by the user — the installer being indistinguishable from a user in this context. The installer should have a clear responsibility to always bring the `default` IngressController resource.
+
+### Goals
+
+1. Eliminate contention with users/automation trying to replace the default IngressController post-installation.
+2. Clarify the role of the ingress-operator with respect to IngressController resource creation
+3. Clarify the role of the installer with respect to the default IngressController resource
+
+### Non-Goals
+
+
+## Proposal
+
+The proposal spans two components: the ingress-operator, and the installer.
+
+The ingress-operator should be modified to stop creating the `default` IngressController resource.
+
+Currently, the installer only renders the `default` IngressController resource when the cluster is configured as "internal". The installer should be changed to _always_ render the `default` IngressController.
+
+### User Stories
+
+As an administrator, I want to change the `default` IngressController from public to private as a day-2 operation.
+
+As an administrator, I want to change the `default` IngressController publishing strategy from LoadBalancerService to HostNetwork.
+
+### Implementation Details/Notes/Constraints [optional]
+
+
+### Risks and Mitigations
+
+
+## Design Details
+
+### New API Field
+
+
+### Migration of existing clusters
+
+
+
+#### Rollbacks
+
+
+### Test Plan
+
+
+#### Migration
+
+### Graduation Criteria
+
+### Upgrade / Downgrade Strategy
+
+
+### Version Skew Strategy
+
+## Implementation History
+
+## Drawbacks
+
+Same as [Route Admission Policy](https://github.com/openshift/enhancements/blob/master/enhancements/ingress/openshift-route-admission-policy.md#drawbacks)
+
+## Alternatives
+
+TODO: Package the default IngressController manifest annotated with `release.openshift.io/create-only=true` into the ingress-operator payload.
```

Comment:
Implemented (but not merged) here:  https://github.com/openshift/cluster-ingress-operator/pull/215

Reply:
What I'd like to understand about that approach is: what is the exhaustive list of circumstances/states in which the the CVO-based resource will be created? If the user deletes the default instance, when is it possible the CVO will attempt to re-create the instance?

In effect, with the CVO approach, does the CVO now race with the user in any case?

Reply:
> If the user deletes the default instance, when is it possible the CVO will attempt to re-create the instance?
> 
> In effect, with the CVO approach, does the CVO now race with the user in any case?

Yeah, I think you are right—it looks like this alternative would just change it so the user would be racing with cluster-version-operator instead of cluster-ingress-operator:  https://github.com/openshift/cluster-version-operator/blob/749ccb062a2947871bf027f27204c1081d4cc822/pkg/cvo/internal/generic.go#L44-L47

Reply:
Added these details to the section.